ios/Classes/CompassController.swift

```diff
@@ -4,36 +4,33 @@ import UIKit
+        var compass = mapView.ornaments.options.compass
         switch settings.position?.value {
         case .BOTTOM_LEFT:
-            mapView.ornaments.options.compass.position = .bottomLeading
-            mapView.ornaments.options.compass.margins = CGPoint(x: (settings.marginLeft?.CGFloat ?? 0.0)/UIScreen.main.scale, y: (settings.marginBottom?.CGFloat ?? 0.0)/UIScreen.main.scale)
+            compass.position = .bottomLeading
+            compass.margins = CGPoint(x: (settings.marginLeft?.CGFloat ?? 0.0)/UIScreen.main.scale, y: (settings.marginBottom?.CGFloat ?? 0.0)/UIScreen.main.scale)
         case .BOTTOM_RIGHT:
-            mapView.ornaments.options.compass.position = .bottomTrailing
-            mapView.ornaments.options.compass.margins = CGPoint(x: (settings.marginRight?.CGFloat ?? 0.0)/UIScreen.main.scale, y: (settings.marginBottom?.CGFloat ?? 0.0)/UIScreen.main.scale)
+            compass.position = .bottomTrailing
+            compass.margins = CGPoint(x: (settings.marginRight?.CGFloat ?? 0.0)/UIScreen.main.scale, y: (settings.marginBottom?.CGFloat ?? 0.0)/UIScreen.main.scale)
         case .TOP_LEFT:
-            mapView.ornaments.options.compass.position = .topLeading
-            mapView.ornaments.options.compass.margins = CGPoint(x: (settings.marginLeft?.CGFloat ?? 0.0)/UIScreen.main.scale, y: (settings.marginTop?.CGFloat ?? 0.0)/UIScreen.main.scale)
+            compass.position = .topLeading
+            compass.margins = CGPoint(x: (settings.marginLeft?.CGFloat ?? 0.0)/UIScreen.main.scale, y: (settings.marginTop?.CGFloat ?? 0.0)/UIScreen.main.scale)
         case .TOP_RIGHT, .none:
-            mapView.ornaments.options.compass.position = .topTrailing
-            mapView.ornaments.options.compass.margins = CGPoint(x: (settings.marginRight?.CGFloat ?? 0.0)/UIScreen.main.scale, y: (settings.marginTop?.CGFloat ?? 0.0)/UIScreen.main.scale)
+            compass.position = .topTrailing
+            compass.margins = CGPoint(x: (settings.marginRight?.CGFloat ?? 0.0)/UIScreen.main.scale, y: (settings.marginTop?.CGFloat ?? 0.0)/UIScreen.main.scale)
         }
 
         if let data = settings.image?.data {
-            mapView.ornaments.options.compass.image = UIImage(data: data, scale: UIScreen.main.scale)
+            compass.image = UIImage(data: data, scale: UIScreen.main.scale)
         }
 
         if let visible = settings.enabled {
             if !visible.boolValue {
-                mapView.ornaments.options.compass.visibility = OrnamentVisibility.hidden
-            } else {
-                if settings.fadeWhenFacingNorth?.boolValue ?? true {
-                    mapView.ornaments.options.compass.visibility = OrnamentVisibility.adaptive
-                } else {
-                    mapView.ornaments.options.compass.visibility = OrnamentVisibility.visible
-                }
+                compass.visibility = visible.boolValue ? .adaptive : .hidden
```

Comment:
Hmm is it correct if's sequence?

Reply:
Ah, true, I missed the `fadeWhenFacingNorthPart` 🤦

Reply:
Should be better now